scripts/update-upstream-changelog.py

```
@@ -0,0 +1,133 @@
+    next_release = text.find("\n## ", unreleased_start + len(UNRELEASED_HEADING))
+    unreleased_end = len(text) if next_release == -1 else next_release
+    unreleased = text[unreleased_start:unreleased_end]
+    if unreleased.count(SECTION_HEADING) != 1:
+        raise ValueError("Unreleased must contain exactly one Automated upstream refreshes section")
+    if unreleased.count(MARKER) != 1:
+        raise ValueError("Automated upstream refreshes must contain exactly one ownership marker")
+
+    section_offset = unreleased.index(SECTION_HEADING)
+    marker_offset = unreleased.index(MARKER)
+    if marker_offset < section_offset:
+        raise ValueError("upstream refresh ownership marker must follow its section heading")
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)